addons/web_editor/controllers/main.py

```diff
@@ -506,7 +506,9 @@ def public_render_template(self, args, kwargs):  # pylint: disable=unused-argume
         xmlid = args[0]
         values = len_args > 1 and args[1] or {}
 
-        View = request.env['ir.ui.view']
+        request_context = kwargs.get('context', {})
+        lang = request_context.get('lang', request.env.context.get('lang'))
+        View = request.env['ir.ui.view'].with_context(lang=lang)
```

Comment:
That's unusual to say the least.
First why is `kwargs` not `**kwargs`? That's probably something missed during httpocalypse, or I am misunderstanding something.
If those were `**kwargs`, context would simply be pop from it and this wouldn't be possible, which would be good.
I am not sure to understand how would the context passed be different than the request one, as passing lang in context kwargs in JS will make it impact the request env lang

So, isn't this a noop? I mean that `request.env['ir.ui.view']` is already equal to `View = request.env['ir.ui.view'].with_context(lang=lang)` no?

Reply:
In the end, this was fixed with https://github.com/odoo/odoo/commit/f05adbc8b59c67cbec4847687a1ee07e620c2a7a